### PR TITLE
[IOTDB-6048] Refactor RatisClient related code

### PR DESCRIPTION
**Pull request by SzyWilliam**
Use AutoClosable to refactor RatisClient related usages.

### CODE REVIEW

**Comment by Cpaulyz** [comment]
Bravo work!

File: iotdb-core/consensus/src/main/java/org/apache/iotdb/consensus/ratis/RatisClient.java [hunk 1]
```diff
@@ -92,7 +97,7 @@ public Factory(
 
     @Override
     public void destroyObject(RaftGroup key, PooledObject<RatisClient> pooledObject) {
-      pooledObject.getObject().close();
+      pooledObject.getObject().destroy();
```
[the remark this answers is not in the document]
Comment (SzyWilliam) [reply iotdb-core/consensus/src/main/java/org/apache/iotdb/consensus/ratis/RatisClient.java]: done

File: iotdb-core/consensus/src/main/java/org/apache/iotdb/consensus/ratis/RatisConsensus.java [hunk 2]
```diff
@@ -298,23 +298,14 @@ public ConsensusWriteResponse write(
...
       }
       writeResult = Utils.deserializeFrom(reply.getMessage().getContent().asReadOnlyByteBuffer());
     } catch (Exception e) {
       return failedWrite(new RatisRequestFailedException(e));
-    } finally {
-      if (client != null) {
-        client.returnSelf();
-      }
-      // statistic the time of write remotely
-      RatisMetricsManager.getInstance()
```
[the remark this answers is not in the document]
Comment (SzyWilliam) [reply iotdb-core/consensus/src/main/java/org/apache/iotdb/consensus/ratis/RatisConsensus.java]: done

**Review by OneSizeFitsQuorum — APPROVED** [review]
LGTM

**Comment by lancelly** [comment]
> Use AutoClosable to refactor RatisClient related usages.

It is obvious that you can maintain all the modules👀

**Pull request merged**